src/packageurl/__init__.py

```diff
@@ -518,7 +518,7 @@ def from_string(cls, purl: str) -> "PackageURL":
         # and the namespace in an npm purl is
         # different from others because it starts with `@`
         # so we need to handle this case separately
-        if type == "npm" and path.startswith("@"):
+        if type.lower() == "npm" and path.startswith("@"):
```

Comment:
Lower the type earlier only once, since this must be always lowered. Do it there https://github.com/package-url/packageurl-python/pull/115/files#diff-1ac2b399571df74798b2fbfab357ffa8cce45a332f1e69b6f1ccf86811d44ba6R502